### ISSUE TITLE
`kafka.KafkaConsumer` get stuck without any exception in for loop

### ISSUE THREAD

**Issue by cf020031308** [dpkp/kafka-python#1552]
Earlier today a script using `kafka-python` with **single process** and **single thread** without gevent something got stuck after several days' running.

The code is as the following but only 'ready to receive' was logged with an **empty** assignment set. The 'received' in the for loop wasn't logged. It stuck for nearly three hours and finally worked after several times of reboots.

```python
consumer = kafka.KafkaConsumer(
    'my_topic',
    group_id='cf08',
    bootstrap_servers='wan.mykafka1.com:9092,wan.mykafka2.com:9092',
    auto_offset_reset='earliest',
    consumer_timeout_ms=70000)
logger.info('ready to receive. assignment: %s', str(consumer.assignment()))
for msg in consumer:
    logger.info('received')
```

In the meantime, IO load is heavy on the kafka server, while scripts subscribing other topics and a script written with `pykafka` subscribing the same topic were all consuming as usual.

**Comment by 649197313** [comment]
well, i had the same problem, and i was puzzled by this question for two days. god dame it!

**Comment by dpkp** [comment]
This sounds like it was probably caused by the consumer deadlock bug that was fixed in the 1.4.4 release. Please upgrade and see if that doesn't fix!
